x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -857,17 +780,16 @@ private static boolean isSecurityEnabled(final OperationMode mode, final boolean
      *
      * @return true is the license is compatible, otherwise false
      */
-    public synchronized boolean isCcrAllowed() {
-        final Status currentStatus = status;
-        return currentStatus.active && isCcrAllowedForOperationMode(currentStatus.mode);
+    public boolean isCcrAllowed() {
+        return checkMinimumLicense(OperationMode.PLATINUM, false, true, true);
     }
 
     public static boolean isCcrAllowedForOperationMode(final OperationMode operationMode) {
-        return isPlatinumOrTrialOperationMode(operationMode);
+        return isPlatinumPlusOrTrialOperationMode(operationMode);
     }
 
-    public static boolean isPlatinumOrTrialOperationMode(final OperationMode operationMode) {
-        return operationMode == OperationMode.PLATINUM || operationMode == OperationMode.TRIAL;
+    public static boolean isPlatinumPlusOrTrialOperationMode(final OperationMode operationMode) {
```

Comment:
Could we make this generic too ? Something like `isMinimumLicense(final OperationMode minimumMode, final OperationMode currentMode)` ? The callers of this should know what the minimum allowed op mode is

Reply:
I thought about consolidate this with `checkMinimumLicense` method as well. I didn't do it because this method is *static* and *not* synchronised. Also it is used as a predicate for RemoteClusterLicenseChecker. So it has some ripple effect. 

It will be more feasible if it does not need to be consolidated with `checkMinimumLicense`. I'll give it a go.

PS: I am confused by the inconsistent synchronisation involved for license check. These static methods can possibly run into racing issues. Is this not a concern? If so why other methods require synchronize?

Reply:
This methoid is now refactored to be [isAllowedByOperationMode](https://github.com/elastic/elasticsearch/pull/51864/files#diff-75ffc27a1f769f6d57256b3a4f04babcR790). It is also used by the [isAllowedByLicenseAndSecurity](https://github.com/elastic/elasticsearch/pull/51864/files#diff-75ffc27a1f769f6d57256b3a4f04babcR823) (previously checkMinimumLicense) method. So that's an extra generification.